include/arch/arm/aarch64/arm_mmu.h

```diff
@@ -133,25 +135,30 @@
 /* Region definition data structure */
 struct arm_mmu_region {
 	/* Region Base Physical Address */
-	uint64_t base_pa;
+	uintptr_t base_pa;
 	/* Region Base Virtual Address */
-	uint64_t base_va;
+	uintptr_t base_va;
```

Comment:
I recall one of comments from @andrewboie that `uintptr_t` should be used for physical addresses(`base_pa`) and `void * `should be used for virtual addresses(`base_va`) just for clarity. not sure if this still holds true.

Reply:
It looks like you are right. I'll fix that.

Reply:
Well, it looks like if I fix this the code becomes a nightmare of casting, so I'm more inclined to leave this as is, with `arch_mem_map` gladly accepting the virt address as `void *` but casting that to `uintptr_t` before passing that down to the MMU code. @andrewboie is the "void * for va addresses" a strict rule?

Reply:
Maybe a stupid idea: Is it possible to use `ulong_t` instead ? The length of the variable will be 32 or 64 function of the arch built, and there will probably are less cast (at least in long term).

Reply:
it's not a hard and fast rule. I *do* tend to use `uintptr_t` for physical addresses and `void *` for linear, but if it makes sense to deviate from this (because of casting mess, or you are doing arithmetic on the pointer value, or whatever) go for it.

> Maybe a stupid idea: Is it possible to use `ulong_t` instead ? The length of the variable will be 32 or 64 function of the arch built, and there will probably are less cast (at least in long term).

`uintptr_t` is preferred. It is an integral value which is guaranteed to be able to hold a linear address.